projects/schematics/src/migrations/4_0/constructor-deprecations/data/window-ref.migration.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  ANGULAR_CORE,
+  PLATFORM_ID_STRING,
+  SPARTACUS_CORE,
+  WINDOW_REF,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const WINDOW_REF_MIGRATION: ConstructorDeprecation = {
+  // projects/core/src/window/window-ref.ts
+  class: WINDOW_REF,
+  importPath: SPARTACUS_CORE,
+  deprecatedParams: [],
```

Comment:
The Document would need to be specified as a deprecated params. Syntax for special injection tokens in action can be seen in `BREAKPOINT_SERVICE_MIGRATION`